tasmota/xdsp_15_tm1637.ino

```diff
@@ -132,52 +154,133 @@
                                         "DisplayClock 2"     // 24 hr format
                                         "DisplayClock 0"     // turn off clock
 
+
+
+Commands specific to TM1638
+====================================
+
+  DisplaySetLEDs        bit_array {0-255} 
+
+                               Sets the 8 LEDs (not the digits!) on the TM1638 module to the binary number represented by the input bit_array.
+                               For example, "DisplaySetLEDs 3" would light up the first and second LED (from left), because 3 => 00000011
+
+
+
+  DisplaySetLED         position {0-7}, value {0|1}
+
+                                Sets a specified LED to either ON or OFF.  e.g., "DisplaySetLED 2, 1" would light up the 3rd LED (2 => 3rd position)
+
+
+
+  DisplayButtons
+
+                                Causes the current state of the buttons/LEDs to be returned as a "STAT" message of the following form:
+                                stat/TM1638/RESULT = {"DisplayButtons": {"S1":<s1>, "S2":<s2>, "S3":<s3>, "S4":<s4>, "S5":<s5>, "S6":<s6>, "S7":<s7>, "S8":<s8>, "Array": [<s1>,<s2>,<s3>,<s4>,<s5>,<s6>,<s7>,<s8>], "Byte": <byte>}}
+                                where <s1>, <s2>, ... are the states of each of the buttons/LEDs, and <byte> is the decimal representation 
+                                of the bit-array that constitutes these states. 
+
+                                For example, if the 5th button is pressed to turn ON LED5, and the "DisplayButtons" command is issued, 
+                                the response STAT message will be of the form:
+                                stat/TM1638/RESULT = {"DisplayButtons": {"S1":0, "S2":0, "S3":0, "S4":0, "S5":1, "S6":0, "S7":0, "S8":0, "Array": [0,0,0,0,1,0,0,0], "Byte": 16}}
+                                because 16 => 2^(5-1)
+
+
+
+  Button Functionality (TM1638 only):
+======================================
+  When this driver is initialized with "DisplayType 2" (TM1638), the buttons on the TM1638 module can be used 
+  to toggle the corresponding LEDs.
+
+  In addition, for each button press, a STAT message of the following form is sent:
+
+  stat/<Topic>/RESULT = {"TM1638_BUTTON<x>":<state>}
+
+      where    <x> = button index (starting at 1)
+           <state> = state of the corresponding LED. It can be "ON" or "OFF" 
+
+  For example, if Button3 is pressed, LED3 would light up and a STAT message is sent as follows:
+  
+  stat/TM1638/RESULT = {"TM1638_BUTTON3":"ON"}
+
+  Pressing Button3 again would turn OFF LED3 and a STAT message is sent as follows:
+
+  stat/TM1638/RESULT = {"TM1638_BUTTON3":"OFF"}
+
```

Comment:
As far as I know TM1638 button support has always been present as it emulates additional standard buttons. What I see here is adding a newkind of button just to support the ones from TM1638 without integration in the Tasmota button pool. It is nice playing but not really functional and again adds redundant code.

Reply:
> As far as I know TM1638 button support has always been present as it emulates additional standard buttons. What I see here is adding a newkind of button just to support the ones from TM1638 without integration in the Tasmota button pool. It is nice playing but not really functional and again adds redundant code.

I can try merging the functionality of the TM1638 Sensor (buttons) driver into this TM1638 Display Driver. But that would mean truly redundant code.
Or, would you rather have me remove the buttons functionality altogether from this TM1638 Display Driver?